### PR TITLE
fix(compatibility): fix mongoose 7 compatibility

### PR DESCRIPTION
**Pull request by romain-gilliotte**
Fixes https://github.com/ForestAdmin/forest-express-mongoose/issues/1037

### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)
- [ ] Ensure that Types have been updated according to your changes (if needed)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: test/utils/mongoose-connect.js [hunk 1]
```diff
@@ -1,10 +1,7 @@
 const mongoose = require('mongoose');
 
 async function mongooseConnect() {
-  await mongoose.connect(
-    'mongodb://localhost:27017/forest-test',
-    { useNewUrlParser: true, useUnifiedTopology: true, useFindAndModify: false },
-  );
+  await mongoose.connect('mongodb://localhost:27017/forest-test');
```
Comment (romain-gilliotte) [inline test/utils/mongoose-connect.js:4]: https://mongoosejs.com/docs/migrating_to_6.html#no-more-deprecation-warning-options

File: src/services/search-builder.js [hunk 2]
```diff
@@ -21,17 +21,17 @@ function SearchBuilder(model, opts, params, searchFields) {
...
+        || value.path === model.schema.options.versionKey) {
         return;
       }
 
       const condition = {};
       const searchValue = params.search.replace(/[-[\]{}()*+!<=:?./\\^$|#\s,]/g, '\\$&');
       const searchRegexp = new RegExp(`.*${searchValue}.*`, 'i');
 
-      if (value.instance === 'ObjectID') {
+      if (value.instance === 'ObjectID' || value.instance === 'ObjectId') {
```
Comment (romain-gilliotte) [inline src/services/search-builder.js:32]: https://mongoosejs.com/docs/migrating_to_7.html#objectid-bsontype-now-has-lowercase-d

File: src/utils/field-analyser.js [hunk 3]
```diff
@@ -148,7 +148,7 @@ class FieldAnalyser {
     if (fieldInfo.enumValues && fieldInfo.enumValues.length) {
       return 'Enum';
     }
-    if (fieldInfo.instance === 'ObjectID') {
+    if (fieldInfo.instance === 'ObjectID' || fieldInfo.instance === 'ObjectId') {
```
Comment (romain-gilliotte) [inline src/utils/field-analyser.js:151]: https://mongoosejs.com/docs/migrating_to_7.html#objectid-bsontype-now-has-lowercase-d

File: src/utils/schema.js [hunk 4]
```diff
@@ -21,12 +21,14 @@ const getNestedFieldType = (mongooseSchema, nestedFieldPath) => {
...
+  if (mongooseSchema.tree?.[currentFieldName]) {
     nestedFieldDeclaration = mongooseSchema.tree[currentFieldName];
-  } else if (mongooseSchema.type) {
+  } else if (mongooseSchema.type?.[currentFieldName]) {
     nestedFieldDeclaration = mongooseSchema.type[currentFieldName];
-  } else {
+  } else if (mongooseSchema[currentFieldName]) {
     nestedFieldDeclaration = mongooseSchema[currentFieldName];
+  } else if (mongooseSchema.type?.tree?.[currentFieldName]) {
+    nestedFieldDeclaration = mongooseSchema.type?.tree?.[currentFieldName];
```
Comment (romain-gilliotte) [inline src/utils/schema.js:31]: I added a condition by looking at the object in the debugger.

It makes all tests pass

File: test/tests/adapters/mongoose.test.js [hunk 5]
```diff
@@ -7,11 +7,15 @@ const { Schema } = mongoose;
...
   afterEach((done) => {
     delete mongoose.models.Foo;
-    delete mongoose.modelSchemas.Foo;
     delete mongoose.models.User;
-    delete mongoose.modelSchemas.User;
     delete mongoose.models.Bar;
-    delete mongoose.modelSchemas.Bar;
+
+    // For older mongoose versions
+    if (mongoose.modelSchemas) {
```
Comment (romain-gilliotte) [inline test/tests/adapters/mongoose.test.js:14]: modelSchemas is `undefined` in mongoose 7

**Review by arnaud-moncel — APPROVED** [review]
Nice 💪

**Pull request merged**